lib/tasks/helpers/newrelicyml.rb

```diff
@@ -0,0 +1,142 @@
+    #<%= "\\n# \#{generated_for_user}\\n#" if generated_for_user %>
+    # For full documentation of agent configuration options, please refer to
+    # https://docs.newrelic.com/docs/agents/ruby-agent/installation-configuration/ruby-agent-configuration
+
+    common: &default_settings
+      # Required license key associated with your New Relic account.
+      license_key: <%= license_key %>
+
+      # Your application name. Renaming here affects where data displays in New
+      # Relic. For more details, see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/renaming-applications
+      app_name: <%= app_name %>
+
+      # To disable the agent regardless of other settings, uncomment the following:
+      # agent_enabled: false
+
+      # Logging level for log/newrelic_agent.log; options are error, warn, info, or
+      # debug.
+      log_level: info
+
+      # All of the following configuration options are optional. Review them, and
+      # uncomment or edit them if they appear relevant to your application needs.
+
+  HEADER
+
+  FOOTER = <<~FOOTER
+    # Environment-specific settings are in this section.
+    # RAILS_ENV or RACK_ENV (as appropriate) is used to determine the environment.
+    # If your application has other named environments, configure them here.
+    development:
+      <<: *default_settings
+      app_name: <%= app_name %> (Development)
+
+    test:
+      <<: *default_settings
+      # It doesn't make sense to report to New Relic from automated test runs.
+      monitor_mode: false
+
+    staging:
+      <<: *default_settings
+      app_name: <%= app_name %> (Staging)
+
+    production:
+      <<: *default_settings
+  FOOTER
+
+  def self.get_configs(defaults)
+    final_configs = {}
+
+    defaults.sort.each do |key, value|
+      next if CRITICAL.include?(key) || SKIP.include?(key)
+
+      if public_config?(value)
+        sanitized_description = sanitize_description(value[:description])
+        description = format_description(sanitized_description)
+        default = default_value(key, value)
+        final_configs[key] = {description: description, default: default}
+      end
+    end
+
+    final_configs
+  end
+
+  def self.public_config?(value)
+    value[:public] == true
+  end
+
+  def self.sanitize_description(description)
+    # remove callouts
+    description = description.split("\n").reject { |line| line.match?('</?Callout') }.join("\n")
+    # remove InlinePopover, keep the text inside type
+    description.gsub!(/<InlinePopover type="(.*)" \/>/, '\1')
+    # remove hyperlinks
+    description.gsub!(/\[([^\]]+)\]\([^\)]+\)/, '\1')
+    # remove single pairs of backticks
+    description.gsub!(/`([^`]+)`/, '\1')
+    # removed hrefs links
+    description.gsub!(/<a href="(.*)">(.*)<\/a>/, '\2')
+
+    description
+  end
+
+  def self.format_description(description)
+    # remove leading and trailing whitespace
+    description = description.strip
```

Comment:
Is there a `#strip!` to make this easier?

Reply:
I was getting a frozen string error in my tests when specifically calling `strip!`.